src/Column/AbstractColumn.php

```diff
@@ -133,7 +133,7 @@ protected function configureOptions(OptionsResolver $resolver)
             ->setAllowedTypes('orderField', ['null', 'string'])
             ->setAllowedTypes('searchable', ['null', 'boolean'])
             ->setAllowedTypes('globalSearchable', ['null', 'boolean'])
-            ->setAllowedTypes('filter', ['null', 'array'])
+            ->setAllowedTypes('filter', ['null', AbstractFilter::class])
```

Comment:
I'm not really sure I like using classes during configuration while we mostly use arrays elsewhere.

Can you implement a unit/functional test covering the reintroduced functionality? That makes it easier for us to review the actual cases this enables.

Reply:
I think this is not an improvement, but requirement to make filtering work. Query builder part is expecting column filter to be an instance of AbstractFilter.

@curry684, is there any chance to get this PR merged?

Reply:
I'll have to defer to @shades684 again as the filtering was one of his thingies.